### PR TITLE
CLUT - 35: Add Image Cell To Chat 

### PR DESCRIPTION
**Pull request by Rieon**
open app -> tap first history -> tap episode

### CODE REVIEW

File: viYarn/Chat/model/EpisodeContent.swift [hunk 1]
```diff
@@ -39,7 +45,14 @@ extension EpisodeContent {
                 }
             case "span":
                 guard let message = try? element.text() else { return nil }
-                arrMessages[arrMessages.count - 1].messages.append(message)
+                let chatElement = ChatElement(message: message, tag: element.tagName(), cellID: ChatTableViewCellMessage.cellID)
+                arrMessages[arrMessages.count - 1].messages.append(chatElement)
```
Comment (bsuvorov) [inline viYarn/Chat/model/EpisodeContent.swift]: p0:
arrMessages[arrMessages.count - 1]
to
arrMessages.lastItem or arrMessages.last

File: viYarn/Chat/model/EpisodeContent.swift [hunk 2]
```diff
@@ -39,7 +45,14 @@ extension EpisodeContent {
...
+            case "img":
+                guard let elemImage = try? element.select("img[src]") else { return nil }
+                guard let message = try? elemImage.attr("src") else { return nil }
+                let chatElement = ChatElement(message: message, tag: element.tagName(), cellID: ChatTableViewCellImage.cellID)
+                arrMessages[arrMessages.count - 1].messages.append(chatElement)
```
Comment (bsuvorov) [inline viYarn/Chat/model/EpisodeContent.swift]: p0: use last

File: viYarn/Chat/view/ChatTableViewCellImage.swift [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import Foundation
+import AlamofireImage
+import UIKit
+
+class ChatTableViewCellImage: UITableViewCell, ChatViewCell {
+    
+    static let cellID = "cellImage"
+    
+    var leadingBubbleConstrain: NSLayoutConstraint!
+    var trailingBubbleConstrain: NSLayoutConstraint!
```
Comment (bsuvorov) [inline viYarn/Chat/view/ChatTableViewCellImage.swift]: p0: why not optional?

File: viYarn/Chat/view/ChatTableViewCellImage.swift [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+import Foundation
+import AlamofireImage
+import UIKit
+
+class ChatTableViewCellImage: UITableViewCell, ChatViewCell {
+    
+    static let cellID = "cellImage"
+    
+    var leadingBubbleConstrain: NSLayoutConstraint!
```
Comment (bsuvorov) [inline viYarn/Chat/view/ChatTableViewCellImage.swift]: p0: why not optional?

File: viYarn/Chat/view/ChatTableViewCellImage.swift [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    let bubbleView: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.backgroundColor = UIColor.yellow
+        view.layer.cornerRadius = 10
+        return view
+    }()
+    
+    var isAuthor: Bool! {
+        didSet{
```
Comment (bsuvorov) [inline viYarn/Chat/view/ChatTableViewCellImage.swift]: p0: style: didSet{ => didSet {

File: viYarn/Chat/view/ChatTableViewCellImage.swift [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            bubbleView.backgroundColor = isAuthor == true ? UIColor.lightGray : UIColor.darkGray
```
Comment (bsuvorov) [inline viYarn/Chat/view/ChatTableViewCellImage.swift]: p0: style:
bubbleView.backgroundColor = isAuthor == true ? UIColor.lightGray : UIColor.darkGray
vs
bubbleView.backgroundColor = isAuthor ? UIColor.lightGray : UIColor.darkGray

File: viYarn/Chat/view/ChatTableViewCellImage.swift [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    
+    let bubbleView: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.backgroundColor = UIColor.yellow
+        view.layer.cornerRadius = 10
+        return view
+    }()
+    
+    var isAuthor: Bool! {
```
Comment (bsuvorov) [inline viYarn/Chat/view/ChatTableViewCellImage.swift]: p0:  Bool! =>  Bool

File: viYarn.xcodeproj/xcuserdata/dmitry.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist [hunk 8]
```diff
@@ -26,7 +26,7 @@
             ignoreCount = "0"
             continueAfterRunningActions = "No"
             filePath = "viYarn/Episode/view/EpisodeTableViewCell.swift"
```
Comment (bsuvorov) [inline viYarn.xcodeproj/xcuserdata/dmitry.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist:28]: p0: add this file to .gitignore

File: viYarn/Chat/model/EpisodeContent.swift [hunk 9]
```diff
@@ -10,10 +10,19 @@ import Foundation
...
 struct EpisodeContent {
-    let chatMessages: [(author: String, messages: [String])]
-    let episodeDescription:String
+    let chatMessages: [ChatElement]
+    let episodeDescription: String
+}
+struct ChatElement {
+    let author: String
+    let message: String
+    let isAuthor: Bool
```
Comment (bsuvorov) [inline viYarn/Chat/model/EpisodeContent.swift:19]: p0: question: что значит isAuthor?

**Review by bsuvorov — APPROVED** [review]
LGTM

**Review by FlareKin — APPROVED** [review]
Well done.😋

**Pull request merged**